t/t4216-log-bloom.sh

```diff
@@ -111,18 +111,37 @@ test_expect_success 'git log with wildcard that resolves to a multiple paths doe
 	test_bloom_filters_not_used "-- file*"
 '
 
-test_expect_success 'setup - a commit-graph chain where latest graph does not have bloom filters' '
+test_expect_success 'setup - add commit-graph to the chain with bloom filters' '
 	test_commit c14 A/anotherFile &&
 	test_commit c15 A/B/anotherFile &&
 	test_commit c16 A/B/C/anotherFile &&
+	git commit-graph write --reachable --changed-paths --split
+'
+
+test_expect_success 'git log uses bloom filters if they exist in all commit graphs in the chain.' '
+	test_bloom_filters_used "-- A/B"
+'
+
+test_expect_success 'setup - add commit-graph to the chain without bloom filters' '
+	test_commit c17 A/anotherFile2 &&
+	test_commit c18 A/B/anotherFile2 &&
+	test_commit c19 A/B/C/anotherFile2 &&
 	GIT_TEST_COMMIT_GRAPH_CHANGED_PATHS=0 git commit-graph write --reachable --split
 '
 
-# A bizarre hack of making a load_bloom_filters call inside get_bloom_filters fixes this test, but
-# breaks everything else because it is obviously wrong. 
-# Commenting it out until we can figure out what's going on. 
-# test_expect_success 'JUST SANITY UNTIL THE SPLIT CHANGE SETTLES DOWN. DELETE AFTER.' '
-#	test_bloom_filters_not_used "-- A/B"
-# '
+test_expect_success 'git log does not use bloom filters if the latest graph does not have bloom filters.' '
+	test_bloom_filters_not_used "-- A/B"
+'
+
+test_expect_success 'setup - add commit-graph to the chain with bloom filters' '
+	test_commit c20 A/anotherFile3 &&
+	test_commit c21 A/B/anotherFile3 &&
+	test_commit c22 A/B/C/anotherFile3 &&
+	git commit-graph write --reachable --changed-paths --split
+'
+
+test_expect_success 'git log uses bloom filters if they exist in the latest but not all commit graphs in the chain.' '
+	test_bloom_filters_used "-- A/B"
+'
```

Comment:
This is the critical test, however, it may not be doing what you want. I recommend rebasing on top of `tb/commit-graph-split-merge` so you can use `--split=no-merge` in these tests. If you look closely at this point, I'm guessing that your commit-graph doesn't have as many layers as you want.